lib/policies/oauth2/site.js

```diff
@@ -1,13 +1,25 @@
-'use strict';
-
 const passport = require('passport');
 const path = require('path');
+const services = require('../../services/index');
 
 module.exports.loginForm = (request, response) => response.render(path.join(__dirname, 'views/login'));
 
 module.exports.login = passport.authenticate('local', { successReturnToOrRedirect: '/', failureRedirect: '/login' });
 
-module.exports.logout = (request, response) => {
-  request.logout();
-  response.redirect('/');
+module.exports.logout = (req, res) => {
+  req.logout();
+  if (!req.query.returnTo) {
+    return res.redirect('/'); // TODO: implement settings section for default view path and return redirects
+  }
+  const returnUrl = decodeURI(req.query.returnTo);
+  if (!req.query.clientId) {
+    return res.redirect(returnUrl);
```

Comment:
This is a bit concerning to me. 

What's happening here is basically: If you want to redirect to any URL you have, just do not provide the `clientID` and we're good to go.

This seems to be a way to skip the security check — something that usually you do not want to do. I'd not provide this feature at all. No clientId? You get an error, and I won't log you out.

Also — the clientId should be also verified whether it's still active.

Reply:
replicated from Auth0 behaviour

Reply:
Can you point the documentation link? I just want to check that out.

Reply:
see the ticket and https://auth0.com/docs/logout
there is actually a bigger problem here even with this PR it still works incorrectly 
If you have time you can investigate this 

Gitter conversation text ---

Neil Stevens @StickNitro Feb 20 16:04
@DrMegavolt Ok so in full,
Navigate to application (user not logged in so redirects to http://localhost:8080/oauth2/authorize?response_type=token&clientId=...&redirectUrl=...
User is prompted to login (using oauth2 credentials)
User is prompted to authorize application
User is redirected to the application (redirectUrl along with token which is cached in sessionStorage and used to access the backend)
Click logout in application (this clears the persisted token and redirects user to http://localhost:8080/logout`)
Logout url above currently redirects to /
At this point I manually navigate to my application and get Step 3

Serhii Kuts @DrMegavolt Feb 20 16:07
and it should be step 2 ? (assuming step 1 will happen automatically)

Neil Stevens @StickNitro Feb 20 16:07
Checking in Chrome dev tools there is a cookie `connect.sid’ that is still present after logout and redirect to /, if i manually remove this cookie after Step 6 above and manually navigate to my application then I get prompted to login (effectively starting at Step 1)

Reply:
Ok thanks. I'll see if we can prioritize this and look into.